### DISCUSSION TITLE
Custom extension using React

### DISCUSSION THREAD

**Discussion by xavArtley** [bokeh/bokeh#11543] [General]
I foward the discussion I opened on [discourse](https://discourse.bokeh.org/t/custom-extension-using-react/8338) here:

```
Hello, I tried to make a Custom Bokeh Extension with material-ui wich use react
You can see my attempt at 
``` 
https://github.com/xavArtley/BokehMaterialUI
```
The package can be installed using python setup.py develop
And I provide a small example in the example directory
However it does not work, I end up with the following error
```
![image|690x132](https://discourse-uploads.bokeh.org/original/2X/5/5e710d2dea8b8d4704b227e21d77c3d754f3adff.png)

![image|614x116](https://discourse-uploads.bokeh.org/original/2X/c/c0cfe5d739ee16fe6a6effad183338dcd72422a4.png)
```
I suppose making a custom extension around react using the bokeh compiler engine is not as simple.
Is there a way to bundle correctly this small example?
```

**Comment by mattpap** [comment]
It looks like react needs a bundler plugin to make it work (at least using this entry point). We don't support this pattern of usage. A quick workaround is to import one of the bundles directly. Alternatively you can use `process` module's polyfill (I suppose `process` package on npm), but that will result in both bundles being included, which I'm sure is not what's expected. Support for this pattern could be implemented (as a TypeScript compiler transform).

**Comment by xavArtley** [comment]
Would be great if bokeh compiler allows utilisation of bundler plugins. Even if I find javascript bundling very confusing and frustrating
